src/Umbraco.Core/Runtime/CoreRuntime.cs

```diff
@@ -140,23 +140,25 @@ protected virtual IFactory Boot(IRegister register, DisposableTimer timer)
                 Compose(composition);
 
                 // acquire the main domain - if this fails then anything that should be registered with MainDom will not operate
-                AcquireMainDom(mainDom);
+                if (AcquireMainDom(mainDom))
```

Comment:
The main time it will return false is when it has been signaled.  When the components run and call register - the register checks for _signaled being true and returns immediately without doing the register. So you are partially not doing the register. The app domain  is not the main dom - it should be doing as little as possible and bailing out as early as possible to minimise the chances of causing weird errors. Now i accept there may be unforseen side effects and dev errors - but that is just a sign that other things need cleaning up. I can see both sides of the argument.

Reply:
Sure, but this appdomain is definitely going to serve at least one request. If we just terminate now, those requests will definitely get errors, whereas there's a chance they won't if we let the app bootup. The only thing that will definitely not work is Examine, the rest will continue to operate as normal, except that nucache will be read into memory via calls to the DB.

Reply:
Your call. If you really think a false return should be ignored you should really change the method signature to return void so a reviewer knows all callers should continue processing regardless of whether the main dom was acquired or not - which does seem odd.

Reply:
We can't change the signature = breaking change and perhaps in some cases a caller will want to know if Acquiring was true. In any case, we cannot just explode if we don't Acquire the MainDom, the normal boot process must continue.